src/database/size.ts

```diff
@@ -0,0 +1,87 @@
+import * as logger from "../logger";
+import * as pathLib from "path";
+
+import { ListRemote, RTDBListRemote } from "./listRemote";
+import { RTDBSizeRemote, SizeRemote } from "./sizeRemote";
+import { Queue } from "../throttler/queue";
+
+/*
+ * For flat objects, performance suffers due to "slow-start"
+ * shallow gets. We set a generous initial list batch size
+ * to mitigate this.
+ */
+const LIST_BATCH_SIZE = 32000;
+const TIMEOUT = 50;
+
+export default class DatabaseSize {
+  path: string;
+  sizeEstimate: number;
+
+  listRemote: ListRemote;
+  sizeRemote: SizeRemote;
+
+  private listQueue: Queue<() => Promise<string[]>, string[]>;
+
+  constructor(instance: string, path: string) {
+    this.path = path;
+    this.sizeEstimate = 0;
+    this.sizeRemote = new RTDBSizeRemote(instance);
+    this.listRemote = new RTDBListRemote(instance);
+    this.listQueue = new Queue({
+      name: "list stack",
+      concurrency: 1,
+      retries: 3,
+    });
+  }
+
+  async execute(): Promise<number> {
+    return this.getSubtreeSize(this.path);
+  }
+
+  private async getSubtreeSize(path: string): Promise<number> {
+    let sizeEstimate = 0;
+    try {
+      sizeEstimate = await this.sizeRemote.sizeNode(path, TIMEOUT);
```

Comment:
I would prefer the sunny case do not reach the `catch` block for readability.
Can you return an object `Promise<SizeResult>`?
```
interface SizeResult {
   success: bool
   bytes: number
}
```
If a catch block is reached, it should get bubbled up and kill the program.

Consider throttle this in a `Stack`. (`Queue` may get memory issues.)
You can tune the concurrent to performance, keep database load in mind.
Linear is not a bad option if our goal is to minimize disruption.